craft_parts/plugins/rust_plugin.py

```diff
@@ -0,0 +1,143 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2022 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The rust plugin."""
+
+import logging
+from textwrap import dedent
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Set, cast
+
+from pydantic import conlist
+
+from . import validator
+from .base import Plugin, PluginModel, extract_plugin_properties
+from .properties import PluginProperties
+
+logger = logging.getLogger(__name__)
+
+# A workaround for mypy false positives
+# see https://github.com/samuelcolvin/pydantic/issues/975#issuecomment-551147305
+if TYPE_CHECKING:
+    UniqueStrList = List[str]
+else:
+    UniqueStrList = conlist(str, unique_items=True)
+
+
+class RustPluginProperties(PluginProperties, PluginModel):
+    """The part properties used by the rust plugin."""
+
+    # part properties required by the plugin
+    rust_features: UniqueStrList = []
+    rust_path: UniqueStrList = ["."]
+    source: str
+
+    @classmethod
+    def unmarshal(cls, data: Dict[str, Any]) -> "RustPluginProperties":
+        """Populate class attributes from the part specification.
+
+        :param data: A dictionary containing part properties.
+
+        :return: The populated plugin properties data object.
+
+        :raise pydantic.ValidationError: If validation fails.
+        """
+        plugin_data = extract_plugin_properties(
+            data, plugin_name="rust", required=["source"]
+        )
+        return cls(**plugin_data)
+
+
+class RustPluginEnvironmentValidator(validator.PluginEnvironmentValidator):
+    """Check the execution environment for the Rust plugin.
+
+    :param part_name: The part whose build environment is being validated.
+    :param env: A string containing the build step environment setup.
+    """
+
+    def validate_environment(self, *, part_dependencies: Optional[List[str]] = None):
+        """Ensure the environment has the dependencies to build Rust applications.
+
+        :param part_dependencies: A list of the parts this part depends on.
+        """
+        for dependency in ["curl", "gcc", "git"]:
```

Comment:
@sergiusens - I moved the dependencies from `get_build_packages()` to `validate_environment()`.  
Is this what you intended for me to do?

Reply:
Please leave gcc in get_build_packages and git. Completely remove curl and you should be verifying that the rustc and cargo commands are available

Reply:
That makes much more sense!